include/flatbuffers/flexbuffers.h

```diff
@@ -26,6 +26,11 @@
 #include <intrin.h>
 #endif
 
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4127) // C4127: conditional expression is constant
```

Comment:
maybe you can also disable 4512 in these files and remove it from CMakeList.txt? That way it will compile for others regardless of their build settings.